test/integration/charms/testapp/src/charm.py

```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env python3
+"""Test app ("db" consumer) charm for Jubilant integration tests"""
```

Comment:
```suggestion
"""Test app ("db" consumer) charm for Jubilant integration tests."""
```
Nit: doesn't ruff normally [complain about this](https://docs.astral.sh/ruff/rules/ends-in-punctuation/)?

Reply:
Probably. I think it's because I have all documentation linting turned off under `test/`, which I think is right. Snippet of `pyproject.toml`:

```
[tool.ruff.lint.per-file-ignores]
"test/*" = [
    # All documentation linting.
    "D",
]
```

Reply:
Ah yes, of course, that explains it, thanks.